packages/sdk/src/evm/functions/contract/index.ts

```diff
@@ -0,0 +1,102 @@
+import { getCompositePluginABI } from "../../common/plugin";
+import { PREBUILT_CONTRACTS_MAP } from "../../contracts";
+import { SmartContract } from "../../contracts/smart-contract";
+import { ContractPublisher } from "../../core/classes/contract-publisher";
+import { getSignerAndProvider } from "../../core/classes/rpc-connection-handler";
+import { ValidContractInstance } from "../../core/types";
+import { AbiSchema } from "../../schema/contracts/custom";
+import {
+  cacheContract,
+  getCachedContract,
+  getCachedStorage,
+  inContractCache,
+} from "../core/cache";
+import { GetContractFromAbiParams, GetContractParams } from "./types";
```

Comment:
instead of breaking out types, and functions let's break the files out by function name and co-locate the types in the same file:

```
functions/
  getContract.ts
  getContractFromAbi.ts
  ...
```

I think that will lead to a cleaner organization and you can always easily import all functions from within other functions by doing `import ... from "./function"`
   

Also I believe `index.ts` is causing a lot of problems for us and we should start moving away from it in favor of real paths and file names and proper named exports

Reply:
Nice I prefer this pattern